packages/ui/components/data-table/index.tsx

```diff
@@ -164,6 +166,7 @@ export function DataTable<TData, TValue>({
               </tr>
             )}
           </TableBody>
+          {tableOverlay && tableOverlay}
```

Comment:
Option to have a tableoverlay to implement custom hovers / anything that needs to be relative to the table.